.github/workflows/full_stack_tests.yml

```diff
@@ -189,105 +189,136 @@ jobs:
           path: desktop_acceptance_test_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
+
       - name: Upload desktop test failure screenshots as artifacts
-        if: ${{ failure() }}
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
           name: desktop-acceptance-test-failure-screenshot-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
           path: /home/runner/work/oppia/oppia_full_stack_test_failure_screenshots/acceptance
+
         # TODO(#22667): We are running this step twice, as snapshot gets deleted if 2nd attempt passes.
-      - name: Uploading diff screenshots as an artifact
-        if: ${{ failure() }}
+      - name: Upload diff screenshots as an artifact
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
-          name: diff-snapshots-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
+          name: diff-snapshots-desktop-attempt-1-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
           # Note: The path must be kept in sync with the one specified in
           # core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts
           path: /home/runner/work/oppia/oppia/core/tests/puppeteer-acceptance-tests/diff-snapshots
+
+      # Desktop Test (Attempt 2)
       - name: Run Desktop Acceptance Test ${{ matrix.suite.name }} (Attempt 2 with screen recording)
-        if: ${{ failure() }}
+        id: desktop-test-attempt-2
+        if: ${{ steps.desktop-test-attempt-1.outcome == 'failure' }}
         run: >
           set -o pipefail;
           VIDEO_RECORDING_IS_ENABLED=1
           xvfb-run -a --server-args="-screen 0, 1920x1080x24"
           python -m scripts.run_acceptance_tests
           --skip-build --suite=${{ matrix.suite.name }} --prod_env --server_log_level=info
           | tee desktop_acceptance_test_retry_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
+
       - name: Upload desktop acceptance retry test logs
-        if: ${{ failure() }}
+        if: ${{ steps.desktop-test-attempt-2.outcome == 'failure' }}
         uses: actions/upload-artifact@v4
         with:
           name: desktop-acceptance-test-retry-logs-${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}
           path: desktop_acceptance_test_retry_${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}.log
-      - name: Uploading Desktop test recordings as Artifacts
-        if: ${{ failure() }}
+      
+      - name: Upload desktop test recordings as an artifact
+        # We are checking for failure of desktop-test-attempt-1 and not desktop-test-attempt-2 because
+        # because we want to upload video recording regardless of whether desktop-test-attempt-2 passes or fails.
```

Comment:
Duplicate "because".

Also I think you want to upload the video for attempt 1 if attempt 1 fails, right? In that case attempt 2 is mainly useful for testing if the error is a flake or a permanent issue.

Reply:
Okay. Got it. Considering above comment, should I add screen recording in the first attempt right?

If we add screen recording in Attempt 1, it will be automatically uploaded here. Though it got me thinking, we can move this step after Mobile Attempt 2, so all test recordings are uploaded in one go? Should I go with it?

Reply:
I would upload it as soon as it's available, then. Just in case there's a failure at some later stage, then we'll still have the recording (hopefully).

More generally I would prioritize making sure we have the debug information whatever happens.

Reply:
Done.